walletconnect-wallet-standard/src/index.js

```diff
@@ -73,265 +63,477 @@ const getConnectParams = (chains) => ({
+        version: "1.0.0",
+        on: this.#on.bind(this),
+      },
+    };
+  }
+
+  // TODO: implement querying the WalletConnect for its supported tx versions
+  /** @returns {import("@solana/wallet-standard-features").SolanaSignTransactionFeature["solana:signTransaction"]["supportedTransactionVersions"]} */
+  get supportedTransactionVersions() {
+    return ["legacy", 0];
+  }
+
+  /** @returns {NonLocalnetChain} */
+  get defaultChain() {
+    const res = this.#chains[0];
+    if (!res) {
+      throw new NoChainsSetError();
     }
-    throw new ClientNotInitializedError();
+    return res;
   }
 
+  /** @type {WalletConnectClient | undefined} */
+  #client;
+
+  /** @type {WalletConnectModal | undefined} */
+  #modal;
+
+  /** @type {import('@walletconnect/types').SignClientTypes.Options} */
+  #options;
+
+  /** @type {NonLocalnetChain[]} */
+  #chains;
+
+  /** @type {Record<NonLocalnetChain, string | null>} */
+  #topics;
+
+  /** @type {Record<NonLocalnetChain, WalletConnectAccount[]>} */
+  #accounts;
+
+  /** @type {{ [E in import("@wallet-standard/features").StandardEventsNames]: Set<import("@wallet-standard/features").StandardEventsListeners[E]> }} */
+  #listeners;
+
   /**
-   * @returns {string}
+   * @param {SolanaWalletConnectWalletStandardCtorArgs} args
    */
-  get publicKey() {
-    if (this._client && this._session) {
-      const chainIndex = this.chains.indexOf(this._network);
-      const { address } = parseAccountId(
-        this._session.namespaces.solana.accounts[chainIndex],
-      );
-      return address;
+  constructor({ chains, ...options }) {
+    this.#chains = chains ?? ["solana:mainnet"];
+    this.#options = options;
+    this.#topics = {
+      "solana:mainnet": null,
+      "solana:devnet": null,
+    };
+    this.#accounts = {
+      "solana:mainnet": [],
+      "solana:devnet": [],
+    };
+    this.#listeners = {
+      change: new Set(),
+    };
+  }
+
+  /** @type {import("@solana/wallet-standard-features").SolanaSignMessageMethod} */
+  async #signMessage(...inputs) {
+    if (!this.#client) {
+      throw new ClientNotInitializedError();
+    }
+    if (inputs.length === 0) {
+      return [];
+    }
+    // TODO: check if chainId matters
+    const chainId = WalletConnectChainID[this.defaultChain];
+    const signatures = [];
+    for (const input of inputs) {
+      const formatted = signMessagesToWalletConnectFormat(input);
+      // eslint-disable-next-line no-await-in-loop
+      const { signature } = await this.#client.request({
+        chainId,
+        topic: this.#topics[chainId],
+        request: {
+          method: WalletConnectRPCMethods.signMessage,
+          params: formatted,
+        },
+      });
+
+      signatures.push(signature);
     }
-    throw new ClientNotInitializedError();
+    const res = [];
+    for (let i = 0; i < inputs.length; i++) {
+      res.push({
+        signedMessage: inputs[i].message,
+        signature: base64ToUint8Ascii(signatures[i]),
```

Comment:
```suggestion
        signature: base58ToBinary(signatures[i]), // need to import base58_to_binary from base58-js
```

signature seems to be base58 according to https://docs.walletconnect.com/2.0/advanced/rpc-reference/solana-rpc#example-3. Will need to verify with an actual wallet